ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/reloperators/HiveProject.java

```diff
@@ -138,6 +143,18 @@ public boolean isSynthetic() {
 
   @Override
   public RelWriter explainTerms(RelWriter pw) {
+    // Remove this if block after upgrading Calcite to 1.35+
+    if (CALCITE_VERSION < 35 
+        && pw.getDetailLevel() == SqlExplainLevel.ALL_ATTRIBUTES && getDigest().contains("SEARCH")) {
+      return ((HiveProject) this.accept(
+          RexUtil.searchShuttle(
+              new RexBuilder(new JavaTypeFactoryImpl(new HiveTypeSystemImpl())),
+              null,
+              -1
+          )
+      )).explainTerms(pw);
+    }
```

Comment:
I assume this block is to address the problem of serializing SEARCH op to JSON. If that's the case then it may preferable to do the expansion inside `HiveRelOptUtil#toJsonString` in a more generic way that its not operator specific.

Another alternative would be to modify `HiveRelJson` classes and apply the fix as done in Calcite.

Reply:
Applying the fix as done in Calcite is more involved, especially when we are planning to upgrade to 1.38+ soon after this upgrade. I am planning to expand SEARCH operator in `HiveRelJson#toJson` which will be a more generic way to do it, and simpler too.